gtsam/hybrid/tests/testGaussianMixtureFactor.cpp

```diff
@@ -387,103 +390,134 @@ TEST(GaussianMixtureFactor, GaussianMixtureModel2) {
 
 namespace test_two_state_estimation {
 
-/// Create Two State Bayes Network with measurements
-static HybridBayesNet CreateBayesNet(double mu0, double mu1, double sigma0,
-                                     double sigma1,
-                                     bool add_second_measurement = false,
-                                     double prior_sigma = 1e-3,
-                                     double measurement_sigma = 3.0) {
-  DiscreteKey m1(M(1), 2);
-  Key z0 = Z(0), z1 = Z(1);
-  Key x0 = X(0), x1 = X(1);
-
-  HybridBayesNet hbn;
+DiscreteKey m1(M(1), 2);
 
-  auto measurement_model = noiseModel::Isotropic::Sigma(1, measurement_sigma);
-  // Add measurement P(z0 | x0)
-  auto p_z0 = std::make_shared<GaussianConditional>(
-      z0, Vector1(0.0), -I_1x1, x0, I_1x1, measurement_model);
-  hbn.push_back(p_z0);
+void addMeasurement(HybridBayesNet& hbn, Key z_key, Key x_key, double sigma) {
+  auto measurement_model = noiseModel::Isotropic::Sigma(1, sigma);
+  hbn.emplace_shared<GaussianConditional>(z_key, Vector1(0.0), I_1x1, x_key,
```

Comment:
The measurement conditional is not the same as the original but it works in this case because `d=0`.